packages/sdk/connect/src/utils/abi-utils.ts

```diff
@@ -28,26 +28,62 @@ export const signatureToAbiDefinition = (fnSignature: string): ABIDefinition =>
     throw new Error(`${fnSignature} is malformed`)
   }
   const method = matches.groups!.method
-  const args = matches.groups!.args.split(',')
 
-  return {
-    name: method,
-    signature: fnSignature,
-    type: 'function',
-    inputs: args.map((type, index) => {
-      const parts = type
+  const argRegex = /\(((?:[^()]+|\((?:[^()]+|\([^()]*\))*\))*)\)|(\b\w+\b(?:(?:\s+\w+)?))/g
+  const args = [...matches.groups!.args.matchAll(argRegex)].map((match) =>
+    match[1] ? `(${match[1]})` : match[2]
+  )
+
+  const inputs = args.map((arg, index) => {
```

Comment:
could you add some comment above about what is happening here (for future generations)

Reply:
I added comment to the function